### PR TITLE
Shivani aggarwal

### PR DESCRIPTION
**Pull request by sa3664**
Feature implemented:

Created some obstacles randomly on the frame and making the crawler to dodge and reach to the destination.

### CODE REVIEW

File: src/main/java/engine/GameEngine.java [hunk 1]
```diff
@@ -89,11 +96,21 @@ public void keyDown() {
...
+
 	private void movePlayer(int xDiff, int yDiff) {
 		TileType attempedLocation = getTileFromCoordinates(getPlayerXCoordinate() + xDiff,
 				getPlayerYCoordinate() + yDiff);
 		if (attempedLocation.equals(TileType.PASSABLE)) {
 			setPlayer(getPlayerXCoordinate() + xDiff, getPlayerYCoordinate() + yDiff);
+		} else if (attempedLocation.equals(TileType.DESTINATION)) {
+			this.level++;
+			this.levelCreator.createLevel(this, level);
+		} else if (attempedLocation.equals(TileType.NOT_PASSABLE)) {
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: I don't see tests for this?

Comment (sa3664) [reply src/main/java/engine/GameEngine.java]: I have added them now . I was working on fixing something in that feature/tests and its code, hence did not upload that.
I have now uploaded the new files.

File: src/main/java/engine/GameEngine.java [hunk 2]
```diff
@@ -109,8 +113,11 @@ private void movePlayer(int xDiff, int yDiff) {
 			this.level++;
 			this.levelCreator.createLevel(this, level);
 		} else if (attempedLocation.equals(TileType.NOT_PASSABLE)) {
-			// send player to original position
+
 			setPlayer((int) startPosition.getX(), (int) startPosition.getY());
+		} else if (attempedLocation.equals(TileType.WINNER)) {
+			setExit(true);
+			System.out.println("GameWon!!");
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: This line makes me sad. Fundamentally untestable and anti-TDD.

Comment (sa3664) [reply src/main/java/engine/GameEngine.java]: Ahh ! Screwed !

I am working on to fix it .

**Pull request closed without merging**